File: src/gtk/sidebar_button.py
from gi.repository import Gtk
from .host_info import HostInfo

class SidebarButton(Gtk.Button):
    __gtype_name__ = "SidebarButton"

    def __init__(self, **kwargs):
        super().__init__(**kwargs)
        
        # Extra Object Creation
        main_split = HostInfo.main_window.main_split

        # Connections
        main_split.connect("notify::collapsed", lambda *_: self.set_visible(main_split.get_collapsed()))
        self.connect("clicked", lambda *_: main_split.set_show_sidebar(True))

        # Apply
        self.set_icon_name("dock-left-symbolic")
        self.set_tooltip_text(_("Show Sidebar"))

File: src/widgets/sidebar_button.ts
import Gtk from "gi://Gtk?version=4.0"

import { SharedVars } from "../utils/shared_vars.js"
import { GClass, OnSignal, PostInit } from "../gobjectify/gobjectify.js"

@GClass()
export class SidebarButton extends Gtk.Button {
	@PostInit
	#ready(): void {
		this.icon_name = (this.get_direction() === Gtk.TextDirection.RTL
			? "warehouse:sidebar-show-right-symbolic"
			: "warehouse:sidebar-show-symbolic"
		)
		SharedVars.main_window?._split_view.$connect("notify::collapsed", () => this.#on_collapse())
	}

	@PostInit
	#on_collapse(): void {
		this.set_visible(SharedVars.main_window?._split_view?.collapsed ?? false)
	}

	@OnSignal("clicked")
	#on_clicked(): void {
		SharedVars.main_window?._split_view?.set_show_sidebar(true)
	}
}
